js/src/utils/registerTouchInteractions.ts

```diff
@@ -10,16 +22,44 @@ export const registerTouchInteractions = ({
   rive,
   fit,
   alignment,
-}) => {
-  if (!canvas || !stateMachines.length || !renderer || !rive || !artboard) {
+}: TouchInteractionsParams) => {
+  if (
+    !canvas ||
+    !stateMachines.length ||
+    !renderer ||
+    !rive ||
+    !artboard ||
+    typeof window === "undefined"
+  ) {
     return null;
   }
 
-  const mouseCallback = (event) => {
-    const boundingRect = event.currentTarget.getBoundingClientRect();
+  const processEventCallback = (event: MouseEvent | TouchEvent) => {
+    const boundingRect = (
+      event.currentTarget as HTMLCanvasElement
+    ).getBoundingClientRect();
 
-    const canvasX = event.clientX - boundingRect.left;
-    const canvasY = event.clientY - boundingRect.top;
+    let canvasX;
+    let canvasY;
+    if (
+      ["touchstart", "touchmove"].indexOf(event.type) > -1 &&
+      (event as TouchEvent).touches?.length
+    ) {
+      canvasX = (event as TouchEvent).touches[0].clientX - boundingRect.left;
+      canvasY = (event as TouchEvent).touches[0].clientY - boundingRect.top;
+      event.preventDefault();
+    } else if (
+      event.type === "touchend" &&
+      (event as TouchEvent).changedTouches?.length
+    ) {
+      canvasX =
+        (event as TouchEvent).changedTouches[0].clientX - boundingRect.left;
+      canvasY =
+        (event as TouchEvent).changedTouches[0].clientY - boundingRect.top;
+    } else {
+      canvasX = (event as MouseEvent).clientX - boundingRect.left;
+      canvasY = (event as MouseEvent).clientY - boundingRect.top;
+    }
```

Comment:
Nitpick, but I'd extract this out into a function the returns the clientX and clientY based on the event type and then do the math to calculate the canvasX and canvasY with the result, since the logic is the same for all three cases